src/components/ListBox.jsx

```diff
@@ -1,64 +1,67 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
 import { Scrollbars } from 'react-custom-scrollbars'
 
-class ListBox extends React.Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      selected: null,
-    }
-  }
+const ListBox = (props) => {
+  const [ selected, setSelected ] = useState(null)
 
-  componentWillReceiveProps(nextProps) {
+  useEffect(() => {
     // Consumers can either pass selected item via props, or let the component
     // to handle it on its own. The former might be useful when you want to
     // extend behaviour (e.g. some ListBoxWithSearchbar), while the latter -
     // when you want a standalone ListBox component.
-    let selected = nextProps.selected || this.state.selected
+    let select = props.selected || selected
 
     // If selected item is not a part of new items, aggressively fallback to
     // first item from the list. We're doing it to be protected from cases
     // when nothing is selected.
-    if (!selected || !nextProps.items.find(item => item.value === selected)) {
-      selected = nextProps.items.get(0)
-      if (!selected) return
-      selected = selected.value
-      nextProps.onClick(selected)
-      this.setState({ selected })
+    if (!select || !props.items.find(item => item.value === select)) {
+      select = props.items.get(0)
+      if (!select) {
```

Comment:
I am just wondering if we can get rid of nested if statement here? we can probably modify line 17

`if ((!select || !props.items.find(item => item.value === select)) && props.items.get(0)) {...}`

Reply:
got rid of that nested statement